### PR TITLE
feat: adds implementation parts to This System component in markdown

### LINKED ISSUE #1527: Add Implementation part headers by default for main component in SSP Markdown
## Issue description / feature objectives

Implementation part headers (e.g `### Implementation part a.`) appear in the SSP Markdown for components when rules are attached. There may be use cases when a high level response is needed at the statement level for the overall system with no rules attached. The main component `This System` is an example of this.

## Caveats / Assumptions

- Assumption - `This System` would be used to provide overall system responses and context so the user would need to be able to provide that at the statement level.
- Assumption - `This System` would not have rules attached

## Completion Criteria

- The `### Implementation part <statement label>` would appear by default for the `This System` component when generating SSP Markdown.

### PR DESCRIPTION
**Pull request by jpower432**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [x] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [X] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [X] My PR meets testing requirements.
- [X] All new and existing tests passed.
- [X] All commits are signed-off.

## Summary

Alters SSP markdown to write the heading for the implementation part if it exists in the control for This System

> Note: This has been tested with the component markdown to ensure the original behavior is not altered and parts are only written when rules are attached

The default behavior when writing SSP is not changed. It can be altered by using the `include-all-parts` flag.
The `--include-all-parts` flag controls `This System` is written in the markdown. When `--include-all-parts` is set, all control parts will be written with the main component present under the `## Implementation for part <part>` section. All other components will still be conditionally added based on the presence of rules.

Closes #1527 

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

**Comment by degenaro** [comment]
Similar comment to PR #1534. Is there a place in the docs that describes this feature for users? Otherwise, looks like you have all the bases covered.

**Comment by jpower432** [comment]
Per discussion at today's issue review, this need changes for being accepted. The behavior should be `opt-in` instead of default.

**Comment by jpower432** [comment]
@degenaro @AleJo2995 @vikas-agarwal76 The logic has been updated based on the provided feedback. Please take another looked.

File: trestle/core/control_writer.py [hunk 1]
```diff
@@ -256,6 +266,23 @@ def _add_implementation_response_prompts(
                     self._insert_comp_info(part_label, dic, context)
         self._md_file.new_hr()
 
+    def _skip_part(self, context: ControlContext, part_label: str, comp_dict: CompDict) -> bool:
+        """
+        Check if a part should be skipped based on rules and context.
+
+        Notes: The default logic is to keep part inclusion rules-based. Using the control
```
Comment (degenaro) [inline trestle/core/control_writer.py:273]: Nit: English could be improved?

**Review by degenaro — APPROVED** [review]
LGTM, by inspection.

One nit cited. Plus, I see Sonar coverage is > 97%, which is fine. I normally try for 100% for the PR if it's not too challenging. Definitely not a requirement, though.

File: trestle/core/control_writer.py (hunk 1 above)
Comment (jpower432) [reply trestle/core/control_writer.py:273]: Thanks @degenaro. I can update the docstring to make it more clear.

**Comment by jpower432** [comment]
> LGTM, by inspection.
> 
> One nit cited. Plus, I see Sonar coverage is > 97%, which is fine. I normally try for 100% for the PR if it's not too challenging. Definitely not a requirement, though.

Thanks @degenaro . I will take a look at the new uncovered lines to see how difficult it would be to remedy.

**Comment by jpower432** [comment]
@degenaro Looks like the new uncovered line was actually moved, not added - https://github.com/oscal-compass/compliance-trestle/pull/1536/files#diff-fcc1b0a823415ac243a1edf58a536c3ec86f7e3bbedacda17bf873c90288f9afR163. The reason that it is uncovered is `ControlContext.prompt_responses` is only set to true at the CLI level in the `Component` and `SSP` context. I tried testing with `prompt_responses` set to true with a generated `Catalog` context, but it lead to an error because the `comp_dict` is not set. If this is something we want to test, seems like it would be better to add a safeguard making sure prompt responses is not true unless in an appropriate context.

**Comment by degenaro** [comment]
@jpower432 Still looks good to me, comments were for over achievement. Already approved! Suggest waiting for @vikas-agarwal76 to chime in.

**Review by vikas-agarwal76 — APPROVED** [review]
LGTM

**Pull request merged**